### PR TITLE
Node v0.12.7 ported to Visual Studio 2015

### PR DESCRIPTION
**Pull request by springmeyer**

### CODE REVIEW

File: common.gypi [hunk 1]
```diff
@@ -89,7 +89,7 @@
         ],
         'msvs_settings': {
           'VCCLCompilerTool': {
-            'RuntimeLibrary': 0, # static release
```
Comment (piscisaureus) [inline common.gypi]: Why did you switch to a dll version of the crt?

**Comment by springmeyer** [comment]
Very sorry. I did not meant to create this pull here (against joyent/node), but rather against mapbox/node - purely to keep track of the changes we need to carry (like the dll version).

File: common.gypi (hunk 1 above)
Comment (springmeyer) [inline common.gypi]: @piscisaureus The Mapbox Studio app (https://www.mapbox.com/mapbox-studio/) uses Node. And it depends on node c++ modules that need to be linked to the dll version (and not statically linked to the c++ runtime) like node-mapnik. So, I build node against the same dll to ensure all C++ code is using the same exact one. Anyway, I realize this is not what node core is going to do and I did not mean for this pull to be against joyent/node as per my comment at https://github.com/joyent/node/pull/25780#issuecomment-126083384. Sorry for the noise.

BTW, this pull does have some fixes needed to compile against Visual 2015 which I will - at some point when I have time - extract and provide as fixes separately.

File: Makefile [hunk 2]
```diff
@@ -343,7 +343,7 @@ tar: $(TARBALL)
 $(BINARYTAR): release-only
 	rm -rf $(BINARYNAME)
 	rm -rf out/deps out/Release
-	$(PYTHON) ./configure --prefix=/ --download=all --with-intl=small-icu \
+	$(PYTHON) ./configure --prefix=/ --download=all --with-intl=none \
```
Comment (srl295) [inline Makefile]: I know this is closed, but why `none` out of curiosity? (saves only 4Mb, opposite from how Node binaries ship, turns off Internationalization API…)

**Pull request closed without merging**